massa-ledger-worker/src/ledger_db.rs

```diff
@@ -310,11 +313,31 @@ impl LedgerDB {
         db.put_or_update_entry_value(batch, serialized_key, &bytes_bytecode);
 
         // datastore
-        for (hash, entry) in ledger_entry.datastore {
+        for (key, entry) in ledger_entry.datastore {
```

Comment:
do we do the checks in the speculative ledger as well ? otherwise it would mean that we accept the change but discover that it is broken when finalizing it. But it should return an error right when it is attempted

Reply:
same for bytecode

Reply:
It's already done in the speculative : https://github.com/massalabs/massa/blob/5a2800714098d09acc3287dfee62530deb1a114f/massa-execution-worker/src/speculative_ledger.rs#L540
https://github.com/massalabs/massa/blob/5a2800714098d09acc3287dfee62530deb1a114f/massa-execution-worker/src/speculative_ledger.rs#L283

Reply:
perfetto :)